exchanges/okx/okx_wrapper.go

```diff
@@ -1712,3 +1736,354 @@ func (ok *Okx) GetFundingRates(ctx context.Context, r *fundingrate.RatesRequest)
+		return nil, margin.ErrOriginalPositionMarginRequired
+	}
+	if req.MarginType != margin.Isolated {
+		return nil, fmt.Errorf("%w %v", margin.ErrMarginTypeUnsupported, req.MarginType)
+	}
+	pairFormat, err := ok.GetPairFormat(req.Asset, true)
+	if err != nil {
+		return nil, err
+	}
+	fPair := req.Pair.Format(pairFormat)
+	marginType := "add"
+	amt := req.NewAllocatedMargin - req.OriginalAllocatedMargin
+	if req.NewAllocatedMargin < req.OriginalAllocatedMargin {
+		marginType = "reduce"
+		amt = req.OriginalAllocatedMargin - req.NewAllocatedMargin
+	}
+	if req.MarginSide == "" {
+		req.MarginSide = "net"
+	}
+	r := IncreaseDecreaseMarginInput{
+		InstrumentID: fPair.String(),
+		PositionSide: req.MarginSide,
+		Type:         marginType,
+		Amount:       amt,
+	}
+
+	if req.Asset == asset.Margin {
+		r.Currency = req.Pair.Base.Item.Symbol
+	}
+
+	resp, err := ok.IncreaseDecreaseMargin(ctx, r)
+	if err != nil {
+		return nil, err
+	}
+
+	return &margin.PositionChangeResponse{
+		Exchange:        ok.Name,
+		Pair:            req.Pair,
+		Asset:           req.Asset,
+		AllocatedMargin: resp.Amount.Float64(),
+		MarginType:      req.MarginType,
+	}, nil
+}
+
+// GetFuturesPositionSummary returns position summary details for an active position
+func (ok *Okx) GetFuturesPositionSummary(ctx context.Context, req *order.PositionSummaryRequest) (*order.PositionSummary, error) {
+	if req == nil {
+		return nil, fmt.Errorf("%w PositionSummaryRequest", common.ErrNilPointer)
+	}
+	if req.CalculateOffline {
+		return nil, common.ErrCannotCalculateOffline
+	}
+	if !ok.SupportsAsset(req.Asset) || !req.Asset.IsFutures() {
+		return nil, fmt.Errorf("%w %v", asset.ErrNotSupported, req.Asset)
+	}
+	fPair, err := ok.FormatExchangeCurrency(req.Pair, req.Asset)
+	if err != nil {
+		return nil, err
+	}
+	instrumentType := ok.GetInstrumentTypeFromAssetItem(req.Asset)
+	positionSummaries, err := ok.GetPositions(ctx, instrumentType, fPair.String(), "")
+	if err != nil {
+		return nil, err
+	}
+	var positionSummary *AccountPosition
+	for i := range positionSummaries {
+		if positionSummaries[i].QuantityOfPosition.Float64() <= 0 {
+			continue
+		}
+		positionSummary = &positionSummaries[i]
+		break
+	}
+	if positionSummary == nil {
+		return nil, fmt.Errorf("%w, received '%v', no positions found", errOnlyOneResponseExpected, len(positionSummaries))
+	}
+	marginMode := margin.Isolated
+	if positionSummary.MarginMode == "cross" {
+		marginMode = margin.Multi
+	}
+
+	acc, err := ok.AccountBalance(ctx, "")
+	if err != nil {
+		return nil, err
+	}
+	if len(acc) != 1 {
+		return nil, fmt.Errorf("%w, received '%v'", errOnlyOneResponseExpected, len(acc))
+	}
+	var freeCollateral, totalCollateral decimal.Decimal
+	for i := range acc[0].Details {
+		if acc[0].Details[i].Currency != positionSummary.Currency {
+			continue
+		}
+		freeCollateral = acc[0].Details[i].AvailableBalance.Decimal()
+		totalCollateral = freeCollateral.Add(acc[0].Details[i].FrozenBalance.Decimal())
```

Comment:
```suggestion
		totalCollateral = acc[0].Details[i].EquityOfCurrency.Decimal()
```

For my test with isolated margin (for a ETH-USDT-SWAP perpetual swap contract), FrozenBalance was 0. This change would work well for USDT/USDC collateralised perps, but you'd also need a USD equiv value if a non-stablecoin is used for collateral. So the ultimate solution would be to show both the currency collateral value and the USD equiv value.

Reply:
Done! In addition, I've just added all the OKx fields. I figure we can consider refinement of fields when we add more exchanges, but better to just return the data okx provides

The following is in demo trading
```
{
 "exchange": "Okx",
 "asset": "perpetualswap",
 "pair": {
  "delimiter": "-",
  "base": "ETH",
  "quote": "USDT-SWAP"
 },
 "position_stats": {
  "maintenance_margin_requirement": "6.30828",
  "initial_margin_requirement": "262.84499999999997",
  "mark_price": "1577.07",
  "current_size": "50",
  "average_open_price": "1579.21",
  "recent_pnl": "-1.07000000000005",
  "margin_fraction": "137.9750465203542",
  "free_collateral": "661.9393031173836",
  "total_collateral": "924.78430311738357",
  "frozen_balance": "262.84499999999997",
  "equity_of_currency": "1134.0096364507167",
  "available_equity": "661.9393031173837",
  "cash_balance": "925.8543031173837",
  "discount_equity": "1134.3611794380165",
  "equity_usd": "1134.3611794380165",
  "isolated_equity": "209.2253333333333",
  "isolated_upl": "-1.5760000000000218",
  "margin_ratio": "137.9750465203542",
  "notional_leverage": "0.8526691006128708"
 }
}%          
```